src/renderer/routes/test_sequencer_panel/components/DesignBar.tsx

```diff
@@ -0,0 +1,266 @@
+        <Button
+          data-testid="add-text-button"
+          className="gap-2"
+          variant="ghost"
+          disabled={true}
+        >
+          <HardDriveDownload size={20} className="stroke-muted-foreground" />
+          Load Test Profile From Cloud
+        </Button>
+        */}
+
+        <div className="grow" />
+
+        <code className="inline-flex items-center justify-center p-3 text-sm text-muted-foreground">
+          {" "}
+          Test {summary.numberOfTestRun + "/" + summary.numberOfTest}
+        </code>
+        <code className="inline-flex items-center justify-center p-3 text-sm text-muted-foreground">
+          {" "}
+          Sequence{" "}
+          {summary.numberOfSequenceRun + "/" + summary.numberOfSequence}
+        </code>
+
+        <CyclePanel />
+
+        <div
+          data-cy="app-status"
+          id="app-status"
+          className="text-md m-1 inline-flex h-8 w-40 items-center justify-center rounded-md border font-semibold text-primary transition-colors"
+        >
+          {mapStatusToDisplay[summary.status]}
+        </div>
+      </div>
+      <div className="py-1" />
+    </div>
+  );
+}
+
+// Helper functions
+
+const getNumberOfTest = (data: TestSequenceElement[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.type === "test") count++;
+  });
+  return count;
+};
+
+const getNumberOfTestRun = (data: TestSequenceElement[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (
+      elem.type === "test" &&
+      elem.status != "pending" &&
+      elem.status != "running"
+    )
+      count++;
+  });
+  return count;
+};
+
+const getNumberOfSequence = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable) count++;
+  });
+  return count;
+};
+
+const getNumberOfSequenceRun = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable && elem.status != "pending" && elem.status != "running")
+      count++;
+  });
+  return count;
+};
+
+export const getGlobalStatus = (
+  cycleRuns: TestSequenceContainer[][],
+  sequences: TestSequenceContainer[],
+  data: TestSequenceElement[],
+): StatusType => {
+  // Create a priority list with all the status
+  const priority = {
+    pending: 0,
+    pass: 1,
+    fail: 2,
+    aborted: 3,
+    paused: 4,
+    running: 5,
+  };
+  // Find highest priority in cycle
+  const highestCycle =
+    cycleRuns.length > 0
+      ? cycleRuns
+          .map((el) => getGlobalStatus([], el, []))
+          .reduce((prev, curr) =>
+            priority[prev] > priority[curr] ? prev : curr,
+          )
+      : "pending";
+  if (sequences.length === 0 && data.length === 0) return highestCycle;
+  // Highest in the view
+  const iter = sequences.length > 0 ? sequences : data;
+  const status = iter
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .map((el: any) => el.status)
```

Comment:
whenever you find yourself going to `any` you should ask yourself if you're doing something wrong.
in this case, `TestSequenceElement` could possibly be a `Conditional` which does not have a status field.
you should just filter only the `Test`s in `data` so then you can type this properly (`TestSequenceContainer | Test`)

Reply:
It's because I'm relying on the "duck typing" of sequences, and TestElems, both of them having the status field, might not be the best way to do it since if we change the field in one of those objects, it will break without notice. I'll change it